### PR TITLE
document recommended usage of system blas/lapack

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/LLNL/spack/issues/2067  (more precisely https://github.com/LLNL/spack/issues/2067#issuecomment-255498874 ).

### CODE REVIEW

**Comment by davydden** [comment]
@adamjstewart @citibeth :  any of you guys want to have a quick look at this doc update related to external blas/lapack?

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -746,6 +746,41 @@ there."  This is reasonable for OpenSSL, which has a stable API.
...
+   Blas/Lapack is ``netlib-lapack`` only because it is the only BLAS / LAPACK
+   provider which use standard names for libraries (as opposed to, for example,
+   `libopenblas.so`).
+
+   Although we specify external package in ``/usr``, Spack is smart enough not
+   to add ``/usr/lib`` to RPATHs, where it could cause unrelated system
+   libraries to be used instead of their Spack equivalents. ``usr/bin`` will be
+   present in PATH, however it will have lower precedence compared to paths
+   from other dependencies. Thereby the usage of system's binaries as opposed
+   to their Spack equivalents is avoided.
```
Comment (tgamblin) [inline lib/spack/docs/getting_started.rst]: I'd reword this to "This ensures that binaries in Spack dependencies are preferred over system binaries."

**Comment by tgamblin** [comment]
LGTM aside from minor nitpick

**Comment by davydden** [comment]
> LGTM aside from minor nitpick

fixed.
